src/util.cpp

```diff
@@ -942,6 +942,36 @@ int CHostAddress::Compare ( const CHostAddress& other ) const
     return thisAddr < otherAddr ? -1 : thisAddr > otherAddr ? 1 : 0;
 }
 
+QString CHostAddress::toString ( const EStringMode eStringMode ) const
+{
+    QString strReturn = InetAddr.toString();
+
+    // special case: for local host address, we do not replace the last byte
+    if ( ( ( eStringMode == SM_IP_NO_LAST_BYTE ) || ( eStringMode == SM_IP_NO_LAST_BYTE_PORT ) ) &&
+         ( InetAddr != QHostAddress ( QHostAddress::LocalHost ) ) && ( InetAddr != QHostAddress ( QHostAddress::LocalHostIPv6 ) ) )
+    {
+        // replace last part by an "x"
+        if ( strReturn.contains ( "." ) )
+        {
+            // IPv4 or IPv4-mapped:
+            strReturn = strReturn.section ( ".", 0, -2 ) + ".x";
+        }
+        else
+        {
+            // IPv6
+            strReturn = strReturn.section ( ":", 0, -2 ) + ":xxxx";
```

Comment:
I thought about this when writing up the report. in IPv4 a host is assigned a single IP. Obfuscating on the last octet (/24) is more than sufficient for privacy.

In IPv6, the provider usually provisions at minimum a /64 per customer. The client's CPE will then autoconfigure hosts behind it with a unique IPv6 address out of the /64. I initially thought obfuscating from the fourth nybble (2001_db8_1234_5678_xxxx) might be the way to go, even if there are multiple clients connecting from same network. However, that /64 defines a single ISP customers network.
Not sure how you'll want to define privacy in that context.

Reply:
>  Not sure how you'll want to define privacy in that context.

Yep, that's a general question which is a bit easier to answer for IPv4 (*just drop 1/4 of the entropy*).
If the goal is being unable to have a working (pingable) IP address in the file names, this PR should be enough.
If the remaing parts of the IP address are expected to prevent people from associating this with another (known) full address of a person, this might not be sufficient for IPv6.

However, I feel that answering this question is hard, so my tendency would be to merge this PR as-is as it fixes a bug (wrong `_x`) and improves things (drops one nibble). If someone has a better idea on how to do the obfuscation, we could always implement that in a follow-up PR.